packages/navigation/src/header-height-context/headerHeightContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {
+    PropsWithChildren,
+    createContext,
+    useContext,
+    useEffect,
+    useMemo,
+    useState,
+} from "react";
+import { useHeaderHeight as useReactNavigationHeaderHeight } from "@react-navigation/elements";
+
+const HeaderHeightContext = createContext<{
+    height: number;
+    setHeight?: (h: (height: number) => number) => void;
+}>({
+    height: 0,
+});
+/**
+ * A header height provider that let's you provide the height of the header to components ABOVE React Navigation in the hiearchy.
+ * Useful if you want something to float above all other content, but not block the header.
+ */
+export const HeaderHeightProvider = ({ children }: PropsWithChildren) => {
+    const [height, setHeight] = useState(0);
+
+    const contextValues = useMemo(() => ({ height, setHeight }), [height]);
+    return (
+        <HeaderHeightContext.Provider value={contextValues}>
+            {children}
+        </HeaderHeightContext.Provider>
+    );
+};
+
+/**
+ * Used within mad navigation to provide the header height
+ */
+const _useMadNavigationHeaderHeight = () => {
+    const context = useContext(HeaderHeightContext);
+    return context.setHeight;
+};
+
+/**
+ * Get the header height
+ * @returns {number} header height
+ */
+export const useHeaderHeight = () => {
+    const context = useContext(HeaderHeightContext);
+    return context.height;
+};
+
+/**
+ * Used within mad navigation to provide the header height
+ */
+export const HeightSender = () => {
```

Comment:
Is there a reason this can't be a hook instead of a component?

Reply:
Yes! We add this to the custom render function of the decorator. Since the render function is a render function and not a component, we can't use hooks inside it.

Reply:
Ahh I see 👍